pilot/pkg/networking/util/util.go

```diff
@@ -435,6 +435,30 @@ func AddSubsetToMetadata(md *core.Metadata, subset string) {
 	}
 }
 
+// AddALPNOverrideToMetadata adds whether the the ALPN prefix should be added to the header
+// to the given core.Metadata struct, if metadata is not initialized, build a new metadata.
+func AddALPNOverrideToMetadata(metadata *core.Metadata, alpnOverride bool) *core.Metadata {
```

Comment:
can we just only include the metadata if alpnOverride is true (or false, I am not sure which one means "do not set it")?

Its not free to add the metadata, actually surprisingly high cost

Reply:
Added return metadata if alpnOverride is true

Reply:
added logic to configureALPNOverride